src/delaunay.js

```diff
@@ -56,14 +52,9 @@ export default class Delaunay {
     if (d.hull && d.hull.length > 2 && collinear(d)) {
       this.collinear = Int32Array.from({length: points.length / 2}, (_, i) => i)
         .sort((i, j) => points[2 * i] - points[2 * j] || points[2 * i + 1] - points[2 * j + 1]); // for exact neighbors
-      const e = this.collinear[0], f = this.collinear[this.collinear.length - 1],
-        bounds = [ points[2 * e], points[2 * e + 1], points[2 * f], points[2 * f + 1] ],
-        r = 1e-8 * Math.hypot(bounds[3] - bounds[1], bounds[2] - bounds[0]);
-      for (let i = 0, n = points.length / 2; i < n; ++i) {
-        const p = jitter(points[2 * i], points[2 * i + 1], r);
-        points[2 * i] = p[0];
-        points[2 * i + 1] = p[1];
-      }
+      const e = this.collinear[0] << 1, f = this.collinear[this.collinear.length - 1] << 1;
+      const r = 1e-9 * Math.hypot(points[f] - points[e], points[f + 1] - points[e + 1]);
```

Comment:
Is it guaranteed that the first and last point in `this.collinear` aren’t coincident? Do we need to compute the bounding box of the collinear points intead?

Reply:
Yes, they are sorted just above. e and f are the opposite extrema.